main.py

```diff
@@ -142,10 +141,21 @@ def main():
             model = Model(config, word_dict, classes)
         model.train(datasets['train'], datasets['val'])
         model.load_best()
-        if 'test' in datasets:
-            test_loader = data_utils.get_dataset_loader(config, datasets['test'], model.word_dict, model.classes, train=False)
-            evaluate(config, model, test_loader, split='test', dump=True)
+
+    if 'test' in datasets:
+        test_loader = data_utils.get_dataset_loader(config, datasets['test'], model.word_dict, model.classes, train=False)
+        test_metrics = evaluate(model, test_loader, config.monitor_metrics)
+        metric_dict = test_metrics.get_metric_dict(use_cache=False)
+        dump_log(config=config, metrics=metric_dict, split='test')
+        print(test_metrics)
+        if config.save_k_predictions > 0:
+            if not config.predict_out_path:
+                config.predict_out_path = os.path.join(config.result_dir, config.run_name, 'predictions.txt')
+            save_top_k_predictions(model.classes, test_metrics.get_y_pred(
+            ), config.predict_out_path, config.save_k_predictions)
```

Comment:
better to have pred() in line 154?? having ")" in the beginning of 155 is strange